### PR TITLE
 #168984699 Add share an article endpoint

### PR DESCRIPTION
**Pull request by hakoemmy**
#### What does this PR do?
- Add  create an article API endpoint
#### Description of Task to be completed?
- Have the following API endpoint working: ``` POST /api/v2/articles ```
#### How should this be manually tested?
just clone the [repo](https://github.com/hakoemmy/TeamWork.git
) and  follow installation guide in [README](https://github.com/hakoemmy/TeamWork/blob/develop/README.md
) then test this endpoint in postman ``` POST /api/v2/articles```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[PT story link](https://www.pivotaltracker.com/story/show/168984699)
#### Screenshots (if appropriate)
create-article

#### Questions: N/A

### CODE REVIEW

File: server/routes/articleRoute.js [hunk 1]
```diff
@@ -11,39 +11,39 @@ import {
...
-  isEmployee,
-  isArticleReqValid,
-  isTheOwner,
-  articleController.editArticle);
+// router.patch('/articles/:articleId',
+//   isValidContentType,
+//   isEmployee,
+//   isArticleReqValid,
+//   isTheOwner,
+//   articleController.editArticle);
```
Comment (Oluwasegun-AA) [inline server/routes/articleRoute.js]: Please remove all commented lines of code from your PR

File: server/routes/articleRoute.js [hunk 2]
```diff
@@ -11,39 +11,39 @@ import {
...
 
-router.delete('/articles/:articleId',
-  isEmployee,
-  isTheOwner,
-  articleController.deleteArticle);
+// router.delete('/articles/:articleId',
+//   isEmployee,
+//   isTheOwner,
+//   articleController.deleteArticle);
```
Comment (Oluwasegun-AA) [inline server/routes/articleRoute.js]: remove commented lines

File: server/routes/articleRoute.js [hunk 3]
```diff
@@ -11,39 +11,39 @@ import {
...
-  isEmployee,
-  articleController.getAllArticle);
+// router.post('/articles/:articleId/comments',
+//   isValidContentType,
+//   isEmployee,
+//   isCommentReqValid,
+//   commentController.commentOnArticle);
+// router.get('/feeds',
+//   isEmployee,
+//   articleController.getAllArticle);
```
Comment (Oluwasegun-AA) [inline server/routes/articleRoute.js]: remove redundant lines of code

File: server/routes/articleRoute.js [hunk 4]
```diff
@@ -11,39 +11,39 @@ import {
...
 
-router.get('/articles/:articleId',
-  isEmployee,
-  isItThere,
-  articleController.getSpecificArticle);
+// router.get('/articles/:articleId',
+//   isEmployee,
+//   isItThere,
+//   articleController.getSpecificArticle);
```
Comment (Oluwasegun-AA) [inline server/routes/articleRoute.js]: please remove

File: server/routes/articleRoute.js (hunk 1 above)
Comment (hakoemmy) [reply server/routes/articleRoute.js]: Removed

File: server/routes/articleRoute.js (hunk 2 above)
Comment (hakoemmy) [reply server/routes/articleRoute.js]: Removed

File: server/routes/articleRoute.js (hunk 3 above)
Comment (hakoemmy) [reply server/routes/articleRoute.js]: Removed

File: server/routes/articleRoute.js (hunk 4 above)
Comment (hakoemmy) [reply server/routes/articleRoute.js]: Removed

File: server/middleware/isEmployee.js [hunk 5]
```diff
@@ -19,7 +19,8 @@ const isEmployee = (req, res, next) => {
 
   try {
     const { id } = jwt.verify(token, process.env.JWTSECRET);
-    if (!User.isUserExist(id)) {
+    const user = await userModel.select('*', 'id=$1', [id]);
+    if (!user.length) {
       return res.status(UNAUTHORIZED).send({
         status: UNAUTHORIZED,
         error: 'Awww, Snap!..Such kind of access token does not match any employee!',
```
Comment (Oluwasegun-AA) [inline server/middleware/isEmployee.js]: @hakoemmy ,  please create an helper file which handles all the responses either success or error.
the helper files should be such that you pass in the required parameters and it does the needful.
i.e
```js
return responsehandler.success(statusCode, 'message', payload)
```
i.e
```js
return responsehandler.error(404, 'item not found')
```

File: server/controllers/articleController.js [hunk 6]
```diff
@@ -1,17 +1,39 @@
...
+          message: 'article successfully created',
+          data: createdArticle,
+        });
+      } catch (e) {
+        console.log(e);
+        return res.status(SERVER_ERROR).json({
+          status: SERVER_ERROR,
+          error: 'OOps, Internal server error occured.',
+        });
+      }
```
Comment (Oluwasegun-AA) [inline server/controllers/articleController.js]: use a responseHandler helper function

File: server/middleware/isEmployee.js (hunk 5 above)
Comment (hakoemmy) [reply server/middleware/isEmployee.js]: Well detailed. Let me fix this ASAP

File: server/controllers/articleController.js (hunk 6 above)
Comment (hakoemmy) [reply server/controllers/articleController.js]: Well detailed. Let me fix this ASAP

Comment (hakoemmy) [reply server/controllers/articleController.js]: I am done with creating helper class

**Pull request merged**